src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const uniqueValues: any[] = [];
+
+  for (const item of items) {
+    const value = item[key];
+
+    if (!uniqueValues.includes(value)) {
+      uniqueValues.push(value);
+    }
```

Comment:
Checking for unique values using `includes` on an array can lead to performance issues as the number of items grows, because `includes` is an O(n) operation. Since you are already using a `Set` in the suggestion above, you can remove this part of the code.